app/repositories/personaldetails/PersonalDetailsRepository.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.personaldetails
+
+import model.ApplicationStatus
+import model.Exceptions.PersonalDetailsNotFound
+import model.persisted.PersonalDetails
+import reactivemongo.api.DB
+import reactivemongo.bson.{BSONDocument, BSONObjectID}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait PersonalDetailsRepository {
+  def update(appId: String, userId: String, personalDetails: PersonalDetails, newAppStatus: ApplicationStatus.Value): Future[Unit]
+
+  def find(appId: String): Future[PersonalDetails]
+}
+
+class PersonalDetailsMongoRepository(implicit mongo: () => DB)
+  extends ReactiveRepository[PersonalDetails, BSONObjectID]("application", mongo, PersonalDetails.personalDetailsFormat,
+    ReactiveMongoFormats.objectIdFormats) with PersonalDetailsRepository {
+  val PersonalDetailsCollection = "personal-details"
+
+  def update(applicationId: String, userId: String, personalDetails: PersonalDetails,
```

Comment:
Good point. It looks like these guard states are prerequisites and they are not part of evaluation in services. Maybe we should put them hardcoded in repositories instead passing from the service?

```
BSONDocument("applicationId" -> applicationId, "userId" -> userId, "$or" -> BSONArray(BSONDocument("applicationStatus" -> CREATED), BSONDocument("applicationStatus" -> IN_PROGRESS))
```